sprouts.py

```diff
@@ -0,0 +1,246 @@
+    win.blit(text, textRect)
+
+#Updates text box surface - Will use to update score 
+def redraw_box():
+    pygame.draw.rect(win, ss.WHITE, (250, 5, 500, 46), 0)
+    pygame.display.update()
+
+#Displays error messages
+def error_display(error_line, error_dot):
+    if error_line:
+        redraw_box()
+        text_writing('Error: Too close to existing dot', ss.RED, 500, 25, 25)
+    
+    if error_dot:
+        redraw_box()
+        text_writing('Error: Start line on dot', ss.RED, 500, 25, 25)
+
+#Draws a dot anywhere, of any colour
+def draw_dot(position, dot_colour):
+    pygame.draw.circle(win, dot_colour, (position), 10, 10)
+    pygame.display.update()
+
+#Draws 3 random starter dots - Implements (faulty) distance check just in case of random overlap
+def start_sequence():
+    temp_list = []
+    for i in range(NumberStartingDots):
+        random_pos = (random.randint(100,900), random.randint(100,500)) #Manual playing field input to not overlap 'menu box', get too close to sides
+        temp_list.append(random_pos)
+
+    for pos in temp_list: #Checks only for exact same coordinates. Needs updating
+        if pos not in dot_pos:
+            dot_pos.append(pos)
+            draw_dot(pos, ss.RED)
+            print(dot_pos)
+        else:
+            print('Overlap')
+            start_sequence()
+
+    pygame.display.update()
+
+#Distance function
+def distance(dot1, dot2):
+    return math.sqrt((dot1[0] - dot2[0])**2 + (dot1[1] - dot2[1])**2)
+
+#Checks whether placed dot is too close to existing dot
+def check_dots(currentdotposition):
+    for dot in dot_pos:
+        if distance((dot), (currentdotposition)) < 30:
+            return False 
+    return True
+
+#Checks whether line starts on existing dot
+def check_line(start_line):
+    for dot in dot_pos:
+        if distance((dot), (start_line)) < 10:
+            return True
+    return False
+
+#These two check if a new dot is placed on a line. They don't yet work how I want them to. 
+#Calculates distance from point to line (line1 to line2). All arguments are "position" (x, y)
+def dist_point_to_line(line1, line2, point):
+    x0 = point[0]
+    y0 = point[1]
+    x1 = line1[0]
+    y1 = line1[1]
+    x2 = line2[0]
+    y2 = line2[1]
+    px = x2-x1
+    py = y2-y1
+    norm = px*px + py*py
+    u =  ((x0 - x1) * px + (y0 - y1) * py) / float(norm)
+    if u > 1:
+        u = 1
+    elif u < 0:
+        u = 0
+    x = x1 + u * px
+    y = y1 + u * py
+    dx = x - x0
+    dy = y - y0
+    dist = sqrt(dx*dx + dy*dy)
+    return dist
+
+#Check to implement:
+def distance_check():
+    if dist_point_to_line(current_line, (current_line -1), dot_pos) < 80:
+        return True
+    return False
+    
+
+if __name__ == '__main__':
+    sprouts()
+
+
+####TO DO
+#Distance Check Update - random dots 
+
+#Update line end - only at dot location.
+#Only allows three lines per one dot.
+#Line can't cross itself... no idea.
+
+#Keep and assign score. 
+
+#Implement GAME OVER
```

Comment:
for this it might be nice to ask for the player's names in the beginning, so you can not only say PLAYER 2 WINS, but, for example 
GAME OVER
KATHI WINS
maybe thats possible by assigning the names to the scores (e.g. kathiscore instead of playertwoscore in line 23) using a string and in the end, include an if-statement like:
if kathiscore > annescore:
    print ('''GAME OVER
                 KATHI WINS''')
if annescore > kathiscore:
     print ('''GAME OVER
                 KATHI WINS''')
or something like that.

Reply:
Definitely! Will implement